File: Python/page_object/07_sauce_labs/test_base.py
import unittest
from selenium import webdriver


class BaseTestCase(unittest.TestCase):

    # Estos datos deben ser extraidos de nuestra cuenta en SauceLabs
    SAUCE_USERNAME = ''
    SAUCE_KEY = ''

    def setUp(self):
        sauce_url = "http://" + self.SAUCE_USERNAME + ":" + self.SAUCE_KEY +"@ondemand.saucelabs.com:80/wd/hub"

        desired_capabilities = {'platform': "Mac OS X 10.9",
                                'browserName': "chrome",
                                'version': "31",
                                }

        self.driver = webdriver.Remote(desired_capabilities=desired_capabilities, command_executor=sauce_url)
        self.driver.get("http://www.mercadolibre.com.ar/")
        self.driver.maximize_window()

    def tearDown(self):
        self.driver.quit()

File: Java/PageObject/07_SauceLabs/TestBase.java
package saucelabs;

import java.net.MalformedURLException;
import java.net.URL;

import org.junit.After;
import org.junit.Before;
import org.openqa.selenium.Platform;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.RemoteWebDriver;

public class TestBase {
	
	WebDriver driver;
	
	String USERNAME = "";
	String ACCESS_KEY = "";
	
	
	@Before
	public void SetUp() throws MalformedURLException {		
        DesiredCapabilities capabilities = DesiredCapabilities.firefox();
        capabilities.setCapability("version", "17");
        capabilities.setCapability("platform", Platform.XP);

        this.driver = new RemoteWebDriver(
                new URL("http://" + USERNAME + ":" + ACCESS_KEY + "@ondemand.saucelabs.com:80/wd/hub"),
                capabilities);

		driver.manage().window().maximize();
		driver.get("http://demo.magentocommerce.com/");
	}
	
	@After
	public void TearDown() {
		driver.quit();
	}
}
